helix-term/src/commands.rs

```diff
@@ -694,6 +694,7 @@ fn goto_buffer(editor: &mut Editor, direction: Direction) {
 
     let id = *id;
 
+    normal_mode_impl(editor);
```

Comment:
Maybe it makes more sense to move this into `editor.switch` instead of adding this only here?  I imagine the bug could appear for other commands aswell. `editor.switch` is called many times and it seems like a huge footgut for it to cause crahses it the editor is not in normal mode. I think this bug could already be produced by other commands if you created a custom insert mode binding for them. I can't think of a situation where this wouldn't want to be a normal mode after a switch.

This will require moving `normal_mode_impl` to `helix-view` but that should be okay since the function is fairly small. 

What do you think?

Reply:
Yeah good call. I think we currently miss some steps when switching modes in `Editor::focus` that would be resolved by that change as well